libfwupdplugin/fu-common.c

```diff
@@ -2258,6 +2258,34 @@ fu_common_is_cpu_intel (void)
 	return FALSE;
 }
 
+/**
+ * fu_common_is_cpu_amd:
+ *
+ * Uses CPUID to discover the CPU vendor and check if it is AMD.
+ *
+ * Return value: %TRUE if the vendor was AMD.
+ *
+ * Since: 1.5.5
+ **/
+gboolean
+fu_common_is_cpu_amd (void)
```

Comment:
of course, this implies adding fu_common_is_cpu_riscv and fu_common_is_cpu_arm in the future. Maybe something like `fu_common_get_cpu_vendor()` might be better, with a `FuCpuVendor` enum with `FU_CPU_VENDOR_AMD` etc.

Reply:
I think that approach is better, a lot less symbols to maintain for ABI.

Reply:
Agree; fixed.